app/src/main/kotlin/com/softteco/template/utils/HandleApiErrorExtension.kt

```diff
@@ -0,0 +1,29 @@
+package com.softteco.template.utils
+
+import com.softteco.template.R
+import com.softteco.template.data.base.error.ErrorEntity
+import com.softteco.template.data.base.error.Result
+import com.softteco.template.ui.components.SnackBarState
+import kotlinx.coroutines.flow.MutableStateFlow
+
+fun <T> handleApiError(
```

Comment:
I'm not sure the file name is well-chosen, since the function is not actually an extension (in my opinion `HandleApiError` would have been enough)